htdocs/comm/propal/class/propal.class.php

```diff
@@ -2060,8 +2060,11 @@ public function set_date($user, $date, $notrigger = 0)
 
 			$this->db->begin();
 
-			$sql = "UPDATE ".MAIN_DB_PREFIX."propal SET datep = '".$this->db->idate($date)."'";
-			$sql .= " WHERE rowid = ".((int) $this->id)." AND fk_statut = ".self::STATUS_DRAFT;
+			if (isset($this->duree_validite) && !empty($this->fin_validite)) {
```

Comment:
set_date is a low level method:
So if we call it, we must update the field.
conditions must be moved on caller side, they will depends on context of this caller page.

Reply:
@eldy moved conditions on caller side

Reply:
Oops problem in PR, 474 files are modified...

Reply:
@eldy sorry manipulation error I create another PR #21839